### ISSUE TITLE
Build System

### ISSUE THREAD

**Issue by brianchirls** [brianchirls/Seriously.js#31]
- Use grunt.js
- Separate out packaged transforms
- Minify with uglify.js or Google Closure Compiler
- Run unit tests on command line
- Run jshint and unit tests before building
- Build with r.js
- option to build with all/some/none of effect and transform plugins included
- In-browser build tool in addition to command line

**Comment by darkyen** [comment]
I would suggest gulp.js over grunt.js simply because gulp is much more readable.

**Comment by brianchirls** [comment]
Yes, I agree. I'm also playing with webpack. I need to spend some more time
with it to figure out the best way to structure dependencies, declarations,
etc.

**Comment by darkyen** [comment]
Webpack is damn cool because of its ability to load and package arbitary formats but then browserify has transforms. 

// just for the same of suggestion
Currently we use 6to5ify and rely on ES2015 style loading powered by browserify, and it has worked pretty well of a solution for us.
 
The ES2015 syntax also makes both the sides (requireJS and commonJS) happy.

**Comment by ilikejames** [comment]
I just made a pull request https://github.com/brianchirls/Seriously.js/pull/105 adding a simple gulp build script. 

`gulp build`

includes the ability to only include certain effects e.g., 

`gulp build --inc "hue-saturation invert"`
or
`gulp build --exc "sketch"`

There is a also tasks for "test" but currently fails.
And task for "jshint" -- lots of errors.

**Comment by genu** [comment]
Another important and useful point to autogenerate documentation from source. It will remove the need to maintain the API wiki. Once set up, and the code is annotated properly, one less thing to worry about.

**Comment by gabrielstuff** [comment]
@brianchirls how goes with all this improvement ? did you have any time ?

Thanks
